src/05-derivatives/04-functional-derivatives.md

```diff
@@ -0,0 +1,373 @@
+| ---------------- | ---------------------------------------------------------------- |
+| `mean`           | Mean across the temporal/4th dimension of the data               |
+| `std`            | Standard deviation across the temporal/4th dimension of the data |
+| `tsnr`           | Temporal SNR (i.e. mean / std)                                   |
+| `sfs`            | Signal fluctuation sensitivity                                   |
+| `alff`           | Amplitude low frequency fluctuations                             |
+| `falff`          | Fractional amplitude of low frequency fluctuations               |
+| `reho`           | Regional homogeneity (voxelwise only)                            |
+| `dcb`, `dcw`     | Voxelwise degree centrality binary and weighted                  |
+| `ecb`, `ecw`     | Voxelwise eigenvector centrality binary and weighted             |
+| `lfcdb`, `lfcdw` | Local functional connectivity density                            |
+| `vmhc`           | Voxel mirrored homotopic connectivity                            |
+
+The following metadata JSON fields are valid for derivative maps:
+
+| Key name       | Description                                                               | Required for suffix        |
+| -------------- | ------------------------------------------------------------------------- | -------------------------- |
+| BandpassFilter | String describing all relevant parameters of the applied bandpass filter. | `alff`, `falff`            |
+| Neighborhood   | String describing neighborhood for regional measures.                     | `reho`                     |
+| Threshold      | String describing threshold used for determining graph edges.             | `dcb`, `dcw`, `ecb`, `ecw` |
+| Method         | String describing method used to calculate measure.                       | `dcb`, `dcw`, `ecb`, `ecw` |
+
+## Time series and regressors
+
+For the purposes of this section, a time series or regressor is a numeric series
+of values corresponding to volumes in a BOLD series. Time series is specifically
+used to indicate an aggregate time series within an ROI, while regressor is used
+as a more general term indicating series that may be derived from other sources,
+such as head motion estimates or physiological recordings.
+
+Regressors will generally be stored as tables, with a row of column headers
+indicating the name of the regressor. In the case where every voxel has a
+regressor, then the data should be stored in a 4D NIfTI file.
+
+Tabular regressor files MUST be accompanied by a data dictionary in JSON format,
+consistent with the format described in
+[Common princioples](../02-common-principles.md).
+
+Volumetric regressor files MAY be accompanied by a data dictionary in JSON
+format, with OPTIONAL `LongName`, `Description`, `Levels`, `Units`, and
+`TermURL` fields.
+
+### General time series and regressors
```

Comment:
We (@adelavega, @effigies, @tsalo) had a discussion about the differences between the `_regressors` and the `_timeseries` suffixes, and I'm no longer sure that I see the need for two separate suffixes, here.

Is there a reason we should not have regressors labeled with the `_timeseries` suffix ? One potential use case: If we extract a time series from the white matter to use as a confound, it is in fact a time series, even if we _intend_ to use it as a regressor. Delineating these two seems unnecessary, to me.

Reply:
+1 for naming things based on the format of the data, and not the user's intention, whenever possible. If two suffixes share the same structure internally, there's no obvious reason they should have different names.

Reply:
+ 1 for timeseries only from me too

Reply:
Okay. Are there cases where we want `_timeseries` to refer to something that isn't 1-per-TR? If we revert to English and not contorting ourselves to be backwards-compatible with this draft, a time series can have any sampling rate, while a regressor is specifically 1-per-TR. So if there's still value in distinguishing those two types of series, and you agree with the distinction I'm drawing, we could move to make the spec correspond to that.

Or we could just go to `_timeseries.tsv` and drop `_regressors.tsv` altogether.

Reply:
Sorry I missed this follow-up, @effigies !

I'm struggling to think of the use case for a time series with some arbitrary sampling rate that's derived from volumes in the BOLD series. So, personally, I'm still pro dropping `_regressors.tsv` altogether. But if there are specific examples that could illustrate this idea a little more, I'd appreciate thinking through them !

Reply:
I don't really consider my own experience broad enough to categorically rule it out, but no, I don't know of any cases where a non-1/TR derivative is actually useful.

I can *imagine* cases like "band-pass filtering physiological time series", but these perhaps would preserve the `_physio` suffix.

Anyway, I would say my inclination is to drop `_regressors` unless a demonstrable need for it arises.

Reply:
I may be missing something, but why does it matter what the sampling rate is? It's not even clear that we need `physio` or `stim` as separate suffixes (yes, I realize that they're locked in now so we're stuck with them); the defining feature in all cases is that we have a uniformly-sampled timeseries with some fixed sampling rate. What the data contained in these files *mean* seems to me like something that should be described in the metadata—there's no reason to put it in the filename. AFAICT just sticking with `_timeseries` is fine, and I'd maybe even push to have future versions of the BIDS-Raw spec deprecate `_physio` and `_stim`.

Reply:
1-per-TR is not necessarily fixed sampling rate ([clustered volume acquisition](https://doi.org/10.1016/j.neuroimage.2007.04.073)). This is kind of what I've had in my head as the reason for checking whether we have other kinds of time series. As long as these are 1-per-TR, then we can piggyback on BOLD series metadata to unambiguously specify timing.

If we want to eventually move all constant-interval series to `_timeseries` as well, we will need to carefully consider sparse acquisition paradigms. It might be as simple as `"SamplingRate": "TR"`, as we currently do in pybids, but just want to flag this.

Reply:
I like the last proposal. @effigies could you send a PR to this branch with specific changes?

Reply:
Just to be clear what these changes will be:

1. All time series can be described in `_timeseries.tsv`.
1. Mandatory metadata for timeseries files include "SamplingRate", which should be in Hz.
    1. A special value "TR" indicates that the sampled points correspond to the timing of a BOLD series with matching keys.
1. The semantically meaningful subsets of time series currently described may still have indicative suffixes &mdash; which actually resolves the `_physio` and `_stim` question nicely, as these just become more special cases of `_timeseries`.

Is this what you were thinking? Are there objections to this overall or in detail? @emdupre @tyarkoni @edickie

I can give this a shot tomorrow.

Reply:
Big fan of the overall direction, thanks @effigies !

One clarification on point 3: `_physio` and `_stim` are allowed _not just_ because they're semantically meaningful suffixes, but because they're grandfathered in, so to speak. Is that right ? 

So long as it's clear that we're moving towards more standardization (and don't plan to sign off on new, special cases of `_timeseries`), then I'm :+1:

Reply:
Agree with all of the above. While `_physio` and `_stim` may be grandfathered in, I don't think this tiny bit of built-in semantic annotation is a sufficient justification for leaving them in the spec in perpetuity. Can we make a note somewhere to remove them in 2.0—or at least, to bring it up for discussion pre-2.0? I would also like to add a note in the current spec deprecating them and encouraging people to use `_timeseries` for everything.